src/checkers/inference/solver/util/StatisticRecorder.java

```diff
@@ -54,20 +33,112 @@ public static synchronized void recordSingleSolvingTime(long value) {
         satSolvingTime.addAndGet((int) value);
     }
 
-    public static void record(StatisticKey key, Long value) {
-        synchronized (statistic) {
-            if (key.equals(StatisticKey.LOGIQL_PREDICATE_SIZE)) {
+    /**
+     * Adds the given value to the statistics for the given key. If an existing value exists for the
+     * given key, this method stores the sum of the new value and the existing value into the key.
+     *
+     * @param key a statistic key. The key is treated case-insensitive: it will always be considered
+     *            in terms of its lower-case equivalent.
+     * @param value a value
+     */
+    public static void record(String key, long value) {
+        synchronized (statistics) {
+            // always use the lower-case version of the given key
+            key = key.toLowerCase();
+
+            if (statistics.get(key) == null || key.contentEquals("logiql_predicate_size")) {
                 // LogiQL predicate size are fixed for same underlining type
                 // system.
-                statistic.put(key, value);
+                statistics.put(key, value);
             } else {
-                long oldValue = statistic.get(key);
-                statistic.put(key, value + oldValue);
+                long oldValue = statistics.get(key);
+                statistics.put(key, value + oldValue);
```

Comment:
maybe we don't need to sum the stats here, if every parallel solver has a unique solver ID